CURATOR-664: Use getHostString() to build connection string in EnsembleTracker even for wildcard addresses

pantherdd opened the pull request:
This is an additional fix for [CURATOR-638](https://issues.apache.org/jira/browse/CURATOR-638) based on PR #425.

kezhuw commented:
Hi @pantherdd , any reason for this change ? [CURATOR-638](https://issues.apache.org/jira/browse/CURATOR-638) emphasizes preference for **configured** DNS name over resolved IP address which become outdated if quorum server migrated. For configured "0.0.0.0", I think "127.0.0.1" is more suitable than "localhost".

pantherdd commented:
Hi @kezhuw ,
My intention was exactly that, to preserve the configured host name even if `clientAddr` is a wildcard address. Sorry for the potentially misleading test case, I just copied the one above and modified it slightly.
I updated both the code and the tests to clarify the behavior post-change.

kezhuw commented on curator-client/src/main/java/org/apache/curator/utils/Compatibility.java:
```diff
@@ -88,7 +88,7 @@ public static boolean hasAddrField()
     }
 
-    public static String getHostAddress(QuorumPeer.QuorumServer server)
+    public static String getHostString(QuorumPeer.QuorumServer server)
```
I think it is ok for this renaming as `Compatibility` is not intended for external usage .

kezhuw reviewed (approved):
Nice work! Thank for your contribution. I have created a separate jira issue [CURATOR-664](https://issues.apache.org/jira/browse/CURATOR-664) for this. Would you mind reword commit message and pr title ?

pantherdd commented:
> Nice work! Thank for your contribution. I have created a separate jira issue [CURATOR-664](https://issues.apache.org/jira/browse/CURATOR-664) for this. Would you mind reword commit message and pr title ?

Sure thing, done. Unfortunately I can't rename the source branch, but that's probably not very important anyway.

pantherdd commented:
There seems to have been some transient network issue during the [PR workflow](https://github.com/apache/curator/actions/runs/4552169337/jobs/8028078609?pr=452):
> Failed to read artifact descriptor for `javax.activation:activation:jar:1.1.1`: Could not transfer artifact `javax.activation:activation:pom:1.1.1` from/to central ([https://repo.maven.apache.org/maven2](https://repo.maven.apache.org/maven2)): transfer failed for [https://repo.maven.apache.org/maven2/javax/activation/activation/1.1.1/activation-1.1.1.pom](https://repo.maven.apache.org/maven2/javax/activation/activation/1.1.1/activation-1.1.1.pom): Connect to `repo.maven.apache.org:443` [`repo.maven.apache.org/146.75.28.215`] failed: Connection timed out (Connection timed out) -> [Help 1]
